### ISSUE TITLE
Nested markup and orphan characters

### ISSUE THREAD

**Issue by gzagatti** [nvim-orgmode/orgmode#430]
### Describe the bug

With the latest changes to in-line markup, I have noticed a bug related with nested markup and orphan characters.

Take the following expression:

```org
zoo =and *boo and= bar* and *goo zar*
```

We end up with verbatim highlight on `=and *boo and=` and bold on `* and *` which is wrong since `bar*` does not start a bold markup.

This problem is particularly frequent with links like the following:

```org
This is [[https://foobar/foo]] foo/
```

Which highlights as italic `/foobar/foo]] foo/`.

### Steps to reproduce

1. Open a new `org` file.
2. Add the following:

```org
foo =and *foo and= bar* and *foo bar*
```

### Expected behavior

The correct behavior is to have verbatim highlight on `=and *foo and=` and bold on `*foo bar*`.

### Emacs functionality

N/A

### Minimal init.lua

N/A

### OS / Distro

Ubuntu 22.04

### Neovim version/commit

0.8

**Comment by kristijanhusak** [comment]
Emacs itself doesn't really know how to handle some of these when link has some spaces:

![screenshot](https://i.imgur.com/r16t8Kk.png)

But I guess spaces are just not a valid char in a link.

**Comment by gzagatti** [comment]
I think I managed to fix this issue with PR #431.

**Comment by kristijanhusak** [comment]
These issues should be fixed on latest master.
